File: src/TellDontAsk.Tests/Doubles/TestShipmentService.cs
using TellDontAsk.Domain;
using TellDontAsk.Service;

namespace TellDontAsk.Tests.Doubles
{
    public class TestShipmentService : ShipmentService
    {
        private Order shippedOrder = null;

        public Order getShippedOrder()
        {
            return shippedOrder;
        }

        public void ship(Order order)
        {
            this.shippedOrder = order;
        }
    }
}

File: php/tests/Doubles/TestShipmentService.php
<?php

namespace Tests\Pitchart\TellDontAskKata\Doubles;

use Pitchart\TellDontAskKata\Domain\Order;
use Pitchart\TellDontAskKata\Services\ShipmentService;

class TestShipmentService implements ShipmentService
{

    private ?Order $shippedOrder = null;

    public function ship(Order $order)
    {
        $this->shippedOrder = $order;
    }

    public function getShippedOrder(): ?Order
    {
        return $this->shippedOrder;
    }
}